### ISSUE TITLE
Add keymap for closing buffer of file under cursor

### ISSUE THREAD

**Issue by GCrispino** [nvim-tree/nvim-tree.lua#3037]
**Is this a question?**
No

**Can this functionality be implemented utilising API?**
If I understood correctly, no.

**Is your feature request related to a problem? Please describe.**
No.

**Describe the solution you'd like**
I believe it might be useful to add a keymap to close the buffer associated with the tree cursor, if such buffer exists.

If this cannot be done, might be useful to have this implemented in the API so a custom keymap can be added.

**Describe alternatives you've considered**
As mentioned, adding a custom keymap using an API method that can achieve this, if possible. 

**Additional context**
I've done a draft implementation of this in a PR in my own fork of nvim-tree: https://github.com/GCrispino/nvim-tree.lua/pull/1

It adds an API method at `Api.node.close.close_file_buffer` and a keymap to `<C-c>` that calls this method.

If the maintainers believe this is worth adding, I'd be happy to create a PR here for this

**Comment by gegoune** [comment]
> **Can this functionality be implemented utilising API?**
>  If I understood correctly, no.

Is that doing what you are after?

```lua
vim.keymap.set('n', '<C-c>', function()
  local filename = require('nvim-tree.api').tree.get_node_under_cursor().absolute_path
  vim.cmd({ cmd = 'bdelete', args = { filename } })
end, opts('duffer delete'))
```

You will probably want to check `filename` for `nil` though.

**Comment by GCrispino** [comment]
> 
> > **Can this functionality be implemented utilising API?**
> >  If I understood correctly, no.
> 
> Is that doing what you are after?
> 
> ```lua
> vim.keymap.set('n', '<C-c>', function()
>   local filename = require('nvim-tree.api').tree.get_node_under_cursor().absolute_path
>   vim.cmd({ cmd = 'bdelete', args = { filename } })
> end, opts('duffer delete'))
> ```

Yeah I think so!

> 
> You will probably want to check `filename` for `nil` though.

About this, would you know how errors are generally reported to the user on nvim-tree? I'm thinking about this case where `filename` would be `nil`, in which some message like "no buffer found for file <filename>" would have to be printed to the user.

**Comment by alex-courtis** [comment]
That's a great idea; I think I'd be using that. Let's add it to API.

Can we please keep vim nomenclature and add an extra call for wipe:
`Api.node.buffer.delete`
`Api.node.buffer.wipe`

RE error messages, yes please, via notify e.g.
https://github.com/nvim-tree/nvim-tree.lua/blob/610a1c189bdb2b9b936169b2ea9d1838f971fa2b/lua/nvim-tree/actions/fs/remove-file.lua#L91
There will be times when vim won't delete the buffer, and will report a message like `E89: No write since last change for buffer 1 (add ! to override)`

**Comment by alex-courtis** [comment]
> ```lua
> vim.cmd({ cmd = "bdelete", args = { filename } })
> ```

Calling the command as a function is a little more robust:
```lua
vim.cmd.bwipe({ "foo", bang = false })
```

It might also be useful to add an option to the api, something like `{ force = true }`

**Comment by GCrispino** [comment]
Hey @alex-courtis , sounds good, thanks for the feedback! I have some questions about your comments:


> Can we please keep vim nomenclature and add an extra call for wipe:
Api.node.buffer.delete
Api.node.buffer.wipe

Here do you mean doing something like the code excerpt I added [in this comment](https://github.com/nvim-tree/nvim-tree.lua/issues/3037#issuecomment-2562536303), but wrapping that into a new API method in `Api.node.buffer.delete`, and then add another method called `Api.node.buffer.wipe`, which would do pretty much the same thing, except that it calls `bwipe` instead of `bdelete`? Did I get this correctly?

> There will be times when vim won't delete the buffer, and will report a message like E89: No write since last change for buffer 1 (add ! to override)

Oh that's true, I hadn't thought of that. Do you have any thoughts on how to handle this? For example, should this not be handled at all (i.e. just call `bdelete`/`bwipe` if a buffer exists for the given file and let it error if it has pending changes), or should we add a `notify.error` call in case the given file has pending changes (i.e. if the buffer is modified)?

-------

> > ```lua
> > vim.cmd({ cmd = "bdelete", args = { filename } })
> > ```
> 
> Calling the command as a function is a little more robust:
> 
> ```lua
> vim.cmd.bwipe({ filename, bang = false })
> ```
> 
> It might also be useful to add an option to the api, something like `{ force = true }`

Sounds good 👍🏼

**Comment by alex-courtis** [comment]
> Did I get this correctly?

Yup, that would be great. Someone will ask for wipe when they see delete.

> For example, should this not be handled at all (i.e. just call bdelete/bwipe if a buffer exists for the given file and let it error if it has pending changes)

If that works we should be fine; vim will effectively `notify.error`.

If that doesn't work e.g. vim throws an exception, we'll need to wrap it in a pcall and report the error, something like https://github.com/nvim-tree/nvim-tree.lua/blob/1ae1c33ce1f4f600dbb9dd8a15b95fbd215358e3/lua/nvim-tree/events.lua#L38

**Comment by GCrispino** [comment]
@alex-courtis @gegoune FYI I've just created a PR for implementing the mentioned API methods.

It's still set as WIP because I haven't run the make commands yet, but the code is pretty much ready.
